### PR TITLE
fix(jit): add field aliases and args support

### PR DESCRIPTION
**Pull request by laststylebender14**
**Summary:**  
_Briefly describe the changes made in this PR._

**Issue Reference(s):**  
Fixes #... _(Replace "..." with the issue number)_

**Build & Testing:**

- [x] I ran `cargo test` successfully.
- [x] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [x] I have added relevant unit & integration tests.
- [x] I have updated the [documentation] accordingly.
- [ ] I have performed a self-review of my code.
- [x] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcallhq.github.io/tree/develop/docs

### CODE REVIEW

File: src/core/jit/snapshots/tailcall__core__jit__builder__tests__variables.snap [hunk 1]
```diff
@@ -40,11 +41,13 @@ expression: plan.into_nested()
...
                         is_scalar: true,
                         directives: [],
+                        alias: None,
                     },
                 ],
             ),
         ),
         is_scalar: false,
         directives: [],
+        alias: None,
```
Comment (meskill) [inline src/core/jit/snapshots/tailcall__core__jit__builder__tests__variables.snap:51]: let's add some test to validate that alias are set sometimes

File: src/core/jit/synth/synth.rs [hunk 2]
```diff
@@ -179,9 +184,9 @@ where
                         let include = self.include(child);
                         if include {
                             let val = obj.get_key(child.name.as_str());
-
+                            let field_name = child.alias.as_deref().unwrap_or(child.name.as_str());
```
Comment (meskill) [inline src/core/jit/synth/synth.rs:187]: can we add a method to Field to wrap this logic and reuse it?

**Review by meskill — CHANGES_REQUESTED** [review]

**Comment by karatakis** [comment]
@meskill, we can merge this. It fixes all JIT conformance test problems except introspection

**Review by meskill — APPROVED** [review]

**Comment by meskill** [comment]
> @meskill, we can merge this. It fixes all JIT conformance test problems except introspection

I've updated the pr with additional fixes and code cleanup. Please, check those and merge the pr if it's ok. Most of the snapshots changes are related to args ordering fix or new field

**Review by karatakis — APPROVED** [review]
It's okay. The TODOs can be addressed later

**Pull request merged**